src/components/ReleaseService/ReleasePlan/TriggerRelease/SingleSelectField.tsx

```diff
@@ -0,0 +1,305 @@
+import React from 'react';
+import {
+  Select,
+  SelectOption,
+  SelectList,
+  SelectOptionProps,
+  MenuToggle,
+  MenuToggleElement,
+  TextInputGroup,
+  TextInputGroupMain,
+  TextInputGroupUtilities,
+  Button,
+  FormGroup,
+} from '@patternfly/react-core';
+import { TimesIcon } from '@patternfly/react-icons/dist/esm/icons/times-icon';
+import { useField } from 'formik';
+import { getFieldId } from '../../../../shared/components/formik-fields/field-utils';
+import FieldHelperText from '../../../../shared/components/formik-fields/FieldHelperText';
+
+type SelectTypeaheadProps = {
+  options: SelectOptionProps[];
+  placeholder?: string;
+  onOptionSelect?: (value: string | number, content: string | number) => void;
+  name: string;
+  label?: React.ReactNode;
+  labelIcon?: React.ReactNode;
+  helpText?: React.ReactNode;
+  required?: boolean;
+  className?: string;
+  setValue: (value: string) => void;
+};
+
+export const SelectTypeahead: React.FunctionComponent<SelectTypeaheadProps> = ({
+  options,
+  label,
+  labelIcon,
+  helpText,
+  required,
+  className,
+  name,
+  placeholder = 'Select an option',
+  onOptionSelect,
+  setValue,
+}) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [selected, setSelected] = React.useState<string>('');
+  const [inputValue, setInputValue] = React.useState<string>('');
+  const [filterValue, setFilterValue] = React.useState<string>('');
+  const [selectOptions, setSelectOptions] = React.useState<SelectOptionProps[]>(options);
+  const [focusedItemIndex, setFocusedItemIndex] = React.useState<number | null>(null);
+  const [activeItemId, setActiveItemId] = React.useState<string | null>(null);
+  const textInputRef = React.useRef<HTMLInputElement>();
+
+  const NO_RESULTS = 'no results';
```

Comment:
would put it into `src/consts/constants.ts` could be useful in other files as well.